gnocchi/storage/file.py

```diff
@@ -95,15 +101,26 @@ def _store_unaggregated_timeserie(self, metric, data, version=3):
         with open(dest, "wb") as f:
             f.write(data)
 
-    def _get_unaggregated_timeserie(self, metric, version=3):
-        path = self._build_unaggregated_timeserie_path(metric, version)
-        try:
-            with open(path, 'rb') as f:
-                return f.read()
-        except IOError as e:
-            if e.errno == errno.ENOENT:
-                raise storage.MetricDoesNotExist(metric)
-            raise
+    def _get_or_create_unaggregated_timeseries(self, metrics, version=3):
+        ts = {}
+        for metric in metrics:
+            path = self._build_unaggregated_timeserie_path(metric, version)
+            try:
+                with open(path, 'rb') as f:
+                    ts[metric] = f.read()
+            except FileNotFoundError:
+                pass
+            except IOError as e:
+                if e.errno != errno.ENOENT:
+                    raise
```

Comment:
this seems wrong that one metric can cause all metrics to fail. that concern can probably be applied to all drivers. we should really be ensuring any error by one doesn't affect another.